test/pkg.jl

```diff
@@ -141,6 +141,27 @@ temp_pkg_dir() do
     @test endswith(str, string(Pkg.installed("Example")))
     @test isempty(Pkg.dependents("Example"))
 
+
+    #-------
+    # 17364
+    #-------
+    let branch_name = "test-branch",
+        branch_commit = "ba3888212e30a7974ac6803a89e64c7098f4865e"
+
+        # create a branch in Example package
+        LibGit2.with(LibGit2.GitRepo, Pkg.dir("Example")) do repo
+            LibGit2.branch!(repo, branch_name, branch_commit, set_head=false)
```

Comment:
If you aren't going to have time to finish this, please communicate as such. A simple "I'm too busy" would be hugely helpful in allocating resources to put out a release candidate. If you can't get this done, could you at least point to how to make simple local commits via the libgit2 bindings? That's not obvious.